src/backend/distributed/commands/trigger.c

```diff
@@ -727,6 +729,49 @@ ErrorOutForTriggerIfNotSupported(Oid relationId)
 }
 
 
+/*
+ * ErrorIfRelationHasUnsupportedTrigger throws an error if given relation has
+ * a trigger that is not supported by Citus.
+ */
+void
+ErrorIfRelationHasUnsupportedTrigger(Oid relationId)
+{
+	List *relationTriggerList = GetExplicitTriggerIdList(relationId);
+
+	Oid triggerId = InvalidOid;
+	foreach_oid(triggerId, relationTriggerList)
+	{
+		ObjectAddress triggerObjectAddress = InvalidObjectAddress;
+		ObjectAddressSet(triggerObjectAddress, TriggerRelationId, triggerId);
+
+		/* triggers that depend on extensions are not supported */
+		if (ObjectAddressDependsOnExtension(&triggerObjectAddress))
+		{
+			ErrorOutForTriggerDependsOnExtension(GetTriggerNameById(triggerId));
```

Comment:
wrapping only the error message in a function seems convoluted. Probably explains the name.

Would there be a way we can move the `ObjectAddressDependOnExtension` check into this function. If the only logical thing in the function would be the error I'd rather repeat ourselfs and copy the same error to two places.

That said, it might actually be better to copy the error around. I find the detail of the message very confusing for the cases we hit here. It is very accurate when we alter a distributed trigger to make it part of an extension. But it makes zero sense when the trigger is part of an extension while we distributed a table.

Reply:
> Would there be a way we can move the ObjectAddressDependOnExtension check into this function

Given that one of two callers of `ErrorOutForTriggerDependsOnExtension` is `Preprocess..` function, I don't think it's possible to move ObjectAddressDependOnExtension check to there.

>  It is very accurate when we alter a distributed trigger to make it part of an extension. But it makes zero sense when the trigger is part of an extension while we distributed a table.

Got it, I will make the necessary changes.

Reply:
Dropping the errordetail section for the latter.